src/Uno.UI/Controls/NativeCommandBarPresenter.iOS.cs

```diff
@@ -46,11 +70,44 @@ private protected override void OnLoaded()
 			// iOS doesn't automatically update the navigation bar position when the status bar visibility changes.
 			void OnStatusBarChanged(StatusBar sender, object args)
 			{
-				navigationBar.SetNeedsLayout();
-				navigationBar.Superview.SetNeedsLayout();
+				_navigationBar!.SetNeedsLayout();
+				_navigationBar!.Superview.SetNeedsLayout();
 			}
 		}
 
+		CommandBarRenderer RendererFactory() => new CommandBarRenderer(_commandBar!);
+
+		protected override Size MeasureOverride(Size size)
+		{
+			var measuredSize = base.MeasureOverride(size);
+
+			if (_isPhone)
+			{
+				switch (UIDevice.CurrentDevice.Orientation)
+				{
+					// On iPhone, the OS may wrongly report the height of the UINavigationBar:
+					//   - After the creation when it's in landscape mode, it's wrongly
+					//     returning the portrait height as measured height (the width is correct)
+					//   - After the rotation from landscape to portrait, the next measured
+					//     height will be the landscape one. Width is also correct.
+					//
+					// This is a ugly hack to circumvent this annoying OS bug.
+					//
+					// NOTE: those values are been fixed for all iOS versions. Tested on iOS 14.
+					case UIDeviceOrientation.LandscapeLeft:
+					case UIDeviceOrientation.LandscapeRight:
+						measuredSize = new Size(measuredSize.Width, 32);
+						break;
+					case UIDeviceOrientation.Portrait:
+					case UIDeviceOrientation.PortraitUpsideDown:
+						measuredSize = new Size(measuredSize.Width, 44);
+						break;
```

Comment:
Are those values depending on the iOS version ?

Reply:
Not on iPhone. Just on iPad.